diff --git a/yt_dlp/extractor/bilibili.py b/yt_dlp/extractor/bilibili.py
index 8675061..7f87104 100644
--- a/yt_dlp/extractor/bilibili.py
+++ b/yt_dlp/extractor/bilibili.py
@@ -678,9 +678,14 @@ class BiliBiliIE(BilibiliBaseIE):
                 'https://api.bilibili.com/x/web-interface/wbi/view/detail', video_id,
                 note='Downloading redirection URL', errnote='Failed to download redirection URL',
                 query=self._sign_wbi(query, video_id), headers=headers)
-            new_url = traverse_obj(detail, ('data', 'View', 'redirect_url', {url_or_none}))
+            new_url = traverse_obj(detail,
+                                   ('data', 'View', 'redirect_url', {url_or_none}),
+                                   ('data', 'View', 'festival_jump_url', {url_or_none}),
+                                   )
             if new_url and BiliBiliBangumiIE.suitable(new_url):
                 return self.url_result(new_url, BiliBiliBangumiIE)
+            if new_url and self.suitable(new_url):
+                return self.url_result(new_url, self)
             raise ExtractorError('Unable to extract initial state')
 
         if traverse_obj(initial_state, ('error', 'trueCode')) == -403:
